torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -157,7 +158,16 @@ def dequantize(self, output_dtype=None):
 
     @staticmethod
     def _quantized_linear_op(input_tensor, weight_tensor, bias):
-        return _quantized_linear_op(input_tensor, weight_tensor, bias)
+        cls = AffineQuantizedTensor
```

Comment:
this feels really hacky, this should be a static method but its being hacked into a class method

Reply:
I feel this should probably be moved to autoquant since it's only used there

Reply:
we need a consistent way to extent aqt kernels and this is what we aligned on.

Reply:
as is, this just breaks autoquant.

Reply:
> I feel this should probably be moved to autoquant since it's only used there

It was used in aqt, its only used in autoquant in your PR because you removed it from aqt.

Reply:
I think moving this to autoquant is a small change? meaning subclass AffineQuantizedTensor and add this staticmethod, why does this break autoquant?

Reply:
does this has to be called in aqt? I didn't see why we have to do this from autoquant code

Reply:
yeah I understand, I think we can provide some other extension point for autoquant to accommodate this change, if people just keep adding things to `_quantized_linear_op` the op is just not going to be maintainable, also it will not possible for people to add new layout and dispatches to affine quantized tensor outside of torchao library